packages/core/data/mm/world/overworld.yml

```diff
@@ -274,16 +274,16 @@
     "Clock Town North": "true"
   locations:
     "Clock Town Great Fairy": "has(STRAY_FAIRY_TOWN)"
-    "Clock Town Great Fairy Alt": "has(STRAY_FAIRY_TOWN) && (has(MASK_DEKU) || has_mask_goron || has_mask_zora)"
+    "Clock Town Great Fairy Alt": "has(STRAY_FAIRY_TOWN) && (has(MASK_DEKU) || has_mask_goron || has_mask_zora || fd_anywhere)"
 "Clock Tower Roof":
   region: CLOCK_TOWER_ROOF
   events:
     MAGIC: "can_play_time"
   exits:
     "Moon": "can_play(SONG_ORDER) && special(MOON)"
   locations:
-    "Clock Tower Roof Skull Kid Ocarina": "has_weapon_range && can_play_time"
-    "Clock Tower Roof Skull Kid Song of Time": "has_weapon_range && can_play_time"
+    "Clock Tower Roof Skull Kid Ocarina": "(has_weapon_range || (fd_anywhere && has_magic)) && can_play_time"
+    "Clock Tower Roof Skull Kid Song of Time": "(has_weapon_range || (fd_anywhere && has_magic)) && can_play_time"
```

Comment:
unsure why magic would be needed if you could just jump attack him. surely he's tall enough where skull kid being at max height would not matter.